### PR TITLE
Update S3 Bucket Should Have Bucket Policy query for AWS CloudFormation 

### LINKED ISSUE #2116: Update S3 Bucket Should Have Bucket Policy query for AWS CloudFormation
### Platform
AWS CloudFormation

### Provider
AWS

### Description
Adding missing reference handling and new validation, namely, that the reference in a Bucket Policy also doesn't match the name of the resource, instead of just the "BucketName" property.

### PR DESCRIPTION
**Pull request by Ruben-Silva**
Closes #2116

**Proposed Changes**

- Adding missing reference handling;
- Adding new validation, namely, that the reference in a Bucket Policy also doesn't match the name of the resource, instead of just the "BucketName" property.

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/query.rego [hunk 1]
```diff
@@ -5,13 +5,14 @@ CxPolicy[result] {
 	resource.Type == "AWS::S3::Bucket"
 	bucket := resource.Properties.BucketName
 	not bucketName(bucket)
-
+    not bucketName(name)
```
Comment (felipe-avelar) [inline assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/query.rego:8]: This should be duplicated?

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please check comments

File: assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/query.rego (hunk 1 above)
Comment (Ruben-Silva) [reply assets/queries/cloudFormation/s3_bucket_should_have_bucket_policy/query.rego:8]: Yes. One checks the name in the property "BucketName", the other checks the name of resource itself.

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**